etcdserver/server.go

```diff
@@ -602,7 +602,47 @@ func (s *EtcdServer) LeaderStats() []byte {
 
 func (s *EtcdServer) StoreStats() []byte { return s.store.JsonStats() }
 
+func (s *EtcdServer) isReadyToAddNewMember() bool {
+	if !s.cfg.StrictReconfigCheck {
+		// the option -strict-reconfig-check isn't passed to etcd,
+		// allow adding a new member unconditionally
+		return true
+	}
+
+	nmembers := 1
+	nstarted := 0
+
+	for _, member := range s.cluster.members {
+		if len(member.Name) != 0 {
+			nstarted++
+		}
+		nmembers++
+	}
+
+	if nstarted == 1 {
+		// a case of adding a new node to 1-member cluster for restoring cluster data
+		// https://github.com/coreos/etcd/blob/master/Documentation/admin_guide.md#restoring-the-cluster
+
+		plog.Infof("a number of started nodes is 1, this cluster is ready to add a new node")
+		return true
+	}
+
+	nquorum := nmembers/2 + 1
+	plog.Infof("all nodes: %d (quorum: %d), started nodes: %d", nmembers, nquorum, nstarted)
```

Comment:
We have printed the add member log line here: https://github.com/coreos/etcd/blob/master/etcdserver/server.go#L867

It is ok for me to log that it passes strict check. You could judge it by yourself: Run this workflow and read through all logs to see whether it makes sense.